src/shared/components/BandPlot.tsx

```diff
@@ -89,20 +90,24 @@ const BandPlot: FC<Props> = ({
       },
     },
     timeFormat,
+    queries,
   },
   theme,
 }) => {
-  const mainColumnName = useMemo(
-    () =>
-      getMainColumnName(
-        selectedFunctions,
-        upperColumnName,
-        mainColumn,
-        lowerColumnName
-      ),
-    [selectedFunctions, upperColumnName, mainColumn, lowerColumnName]
-  )
-
+  const mainColumnName = useMemo(() => {
+    const aggregateFunctions = get(
+      queries,
+      `${activeQueryIndex}.builderConfig.functions`,
```

Comment:
what happens when a band plot is made using raw flux and not the query builder? feels like these wont be defined?

Reply:
Thanks for pointing out this case. Yes, good point. I will make a change. When selectedFunctions are empty (no aggregate functions are found in the builderConfig) and a "mainColumn" is specified, in which case it should just be the "mainColumn"

Reply:
This is actually a mode problem, Query Builder vs Script Editor. The latest commit allows Band Plot to work in all modes.